client/rest/test/plugins/metadata/metal_spec.js

```diff
@@ -23,309 +23,255 @@ const {
+
+		it('extract chunk with text', () => {
 			// Arrange:
 			const metadata = getMetadata(10);
 			const nextMetadata = getMetadata(20);
-			// Act:
-			const result = metal.extractChunk(Buffer.from(metadata.value));
-			// Assert:
-			expect(result).to.deep.equal({
+			// Act + Assert:
+			expect(metal.extractChunk(Buffer.from(metadata.value))).to.deep.equal({
 				magic: 0x00,
 				scopedMetadataKey: nextMetadata.scopedMetadataKey,
 				chunkPayload: Buffer.from(metadata.value.slice(12)),
 				text: true
 			});
 		});
-		it('extract chunk and returns the value end chunk with text', () => {
+
+		it('extract end chunk with text', () => {
 			// Arrange:
 			const metadata = getMetadata(60);
-			const textSection = new MetalSeal(imageBytes.length, 'image/png', 'image.png', 'test').stringify();
-			const { combinedPayload } = combinePayloadWithText(imageBytes, textSection);
+			const textSection = new MetalSeal(testData.imageBytes.length, 'image/png', 'image.png', 'test').stringify();
+			const { combinedPayload } = combinePayloadWithText(testData.imageBytes, textSection);
 			const checkSum = generateChecksum(combinedPayload);
-
-			// Act:
-			const result = metal.extractChunk(Buffer.from(metadata.value));
-			// Assert:
-			expect(result).to.deep.equal({
+			// Act + Assert:
+			expect(metal.extractChunk(Buffer.from(metadata.value))).to.deep.equal({
 				magic: 0x80,
 				scopedMetadataKey: checkSum,
 				chunkPayload: Buffer.from(metadata.value.slice(12)),
 				text: false
 			});
 		});
-		it('can not extract size is shorten', () => {
-			// Arrange:
-			const metadata = getMetadata(10);
-			const metadataValue = metadata.value.slice(0, 11);
-			// Act:
-			const result = () => metal.extractChunk(Buffer.from(metadataValue));
-			// Assert:
-			expect(result).to.throw(`Invalid metadata value size ${metadataValue.length}`);
-		});
-		it('can not extract size is longer', () => {
-			// Arrange:
-			const metadata = getMetadata(10);
-			const metadataValue = metadata.value.concat(Buffer.alloc(1, 0x00));
-			// Act:
-			const result = () => metal.extractChunk(Buffer.from(metadataValue));
-			// Assert:
-			expect(result).to.throw(`Invalid metadata value size ${metadataValue.length}`);
-		});
-	});
-	describe('generateMetadataKey', () => {
-		it('generate metadata key', () => {
-			// Arrange:
-			const metadata = getMetadata(10);
-			const input = metadata.value;
-			// Act:
-			const result = metal.generateMetadataKey(new Uint8Array(input));
-			// Assert:
-			expect(result).to.deep.equal(metadata.scopedMetadataKey);
-		});
-		it('can not generates metadata key input is empty', () => {
-			// Arrange:
-			const input = Buffer.alloc(0);
-			// Act:
-			const result = () => metal.generateMetadataKey(input);
-			// Assert:
-			expect(result).to.throw('Input must not be empty');
-		});
 	});
+
 	describe('splitChunkPayloadAndText', () => {
 		it('split chunk payload and text', () => {
```

Comment:
would consider descriptions:
can extract payload and text when both present
can extract payload when only payload present

Reply:
🙏